patches/api/0392-Add-Shearable-API.patch

```diff
@@ -77,19 +77,52 @@ index cef1700834643fe28ed5737578d91ecefbe99e2f..86c0043ef4e1288b6fe2f68a9b6d01c3
  
      /**
       * Checks for the presence of custom potion effects to be applied to the
+diff --git a/src/main/java/org/bukkit/entity/Shearable.java b/src/main/java/org/bukkit/entity/Shearable.java
+index 0215d20f81bfbef080f86ce46147a38f71310f65..9967c8a52ddd2c7e10db49a3f166731373f1ba45 100644
+--- a/src/main/java/org/bukkit/entity/Shearable.java
++++ b/src/main/java/org/bukkit/entity/Shearable.java
+@@ -2,20 +2,30 @@ package org.bukkit.entity;
+ 
+ /**
+  * Represents an entity which can be shorn with shears.
++ * @deprecated Spigots shearable API miserably fails at capturing all entities that may be sheared by a player, like
++ * mushroom cows which, once sheared, convert into normal cows. For such entities, methods like
++ * {@link #setSheared(boolean)} or {@link #isSheared()} make no sense, making this API and interface dead API from
++ * the get-go.
+  */
++@Deprecated(forRemoval = true, since = "1.21")
```

Comment:
Isn't this conflating the idea of a shearable state (a sheep will regrow its wool) and a property of entities (a mooshroom can only be sheared once) of an entity?

Reply:
Yes, spigots only exposes entities in which have a property for that, whereas papers exposes all entities which can be sheared, akin to how it's handled internally too

Reply:
I guess it may be worth investigating the need for some RegrowingShearable subtype of the paper interface to cover the ability to "unshear". But yea, nothing that fits nicely into the spigot type definition here, hence deprecated.